lib/utils/deploy.go

```diff
@@ -0,0 +1,144 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	g "github.com/sdslabs/katana/configs"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func DeployChallenge(chall_name, team_name string, patched bool) error {
+
+	team_namespace := team_name + "-ns"
+
+	log.Println(team_namespace)
+	kubeclient, err := GetClient(g.KatanaConfig.KubeConfig)
+	if err != nil {
+		return err
+	}
+
+	deploymentsClient := kubeclient.AppsV1().Deployments(team_namespace)
+	imageName := chall_name + ":latest"
+	if patched {
+		/// Retrieve the existing deployment
+        existingDeployment, err := deploymentsClient.Get(context.TODO(), chall_name, metav1.GetOptions{})
+        if err != nil {
+            fmt.Println("Error in retrieving existing deployment.")
+            log.Println(err)
+            return err
+        }
+
+        existingDeployment.Spec.Template.Spec.Containers[0].Image = team_name + "/" + chall_name
+
+        _, err = deploymentsClient.Update(context.TODO(), existingDeployment, metav1.UpdateOptions{})
+        if err != nil {
+            fmt.Println("Error in updating deployment.")
+            log.Println(err)
+            return err
+        }
+
+        fmt.Println("Updated deployment with new image.")
+        return nil
+	}
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: team_namespace,
+			Name:      chall_name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: int32Ptr(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": chall_name,
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": chall_name,
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:            chall_name + "-" + team_name,
+							Image:           imageName,
+							ImagePullPolicy: v1.PullPolicy("Never"),
+							Ports: []v1.ContainerPort{
+								{
+									Name:          "challenge-port",
+									Protocol:      v1.ProtocolTCP,
```

Comment:
Protocol and port should not be hardcoded, for example, for pwn challenges, we would be using netcat

Reply:
To be fixed in future